autokeras/hypermodel/processor.py

```diff
@@ -255,3 +255,140 @@ def output_types(self):
 
     def output_shape(self):
         return self._shape
+
+
+class ImageAugment(HyperPreprocessor):
+
+    def __init__(self,
+                 rotation_range=None,
+                 whether_random_crop=None,
+                 brightness_range=None,  # fraction 0-1  [X]
+                 saturation_range=None,  # fraction 0-1  [X]
+                 contrast_range=None,  # fraction 0-1  [X]
+                 horizontal_flip=None,  # boolean  [X]
+                 vertical_flip=None,
+                 whether_translation=None,
+                 gaussian_noise=None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.rotation_range = rotation_range
+        self.whether_random_crop = whether_random_crop
+        self.brightness_range = brightness_range
+        self.saturation_range = saturation_range
+        self.contrast_range = contrast_range
+        self.horizontal_flip = horizontal_flip
+        self.vertical_flip = vertical_flip
+        self.whether_translation = whether_translation
+        self.gaussian_noise = gaussian_noise
+        self._shape = None
+
+    @staticmethod
+    def __get_min_and_max(value, name):
+        if isinstance(value, (tuple, list)):
+            if len(value) != 2:
+                raise ValueError(
+                    'Argument %s expected either a float between 0 and 1, '
+                    'or a tuple of 2 floats between 0 and 1, '
+                    'but got: %s' % (value, name))
+            min_value, max_value = value
+        else:
+            min_value = 1. - value
+            max_value = 1. + value
+        return min_value, max_value
+
+    def transform(self, x):
+        self._shape = x.shape
+        rotation_range = self.rotation_range
+        if rotation_range is None:
+            rotation_range = self._hp.Choice('rotation_range')
```

Comment:
Please look into the implementation of hp here: https://github.com/keras-team/keras-tuner/blob/master/kerastuner/engine/hyperparameters.py
You may take a look at the block.py for the usage of the hp.
We need the name, values, defaults all specified when we call it.
Besides Choice, we also have Range, Linear and Fixed. Please select the best one for each hp.